test/beforeRequest.test.js

```diff
@@ -606,43 +539,51 @@ describe("test validating response", () => {
             test(`reloading off after sign: ${element.name}`, () => {
                 let before;
                 let after;
+                const commStruct = {};
+                commStruct[commVersion] = {L: G, H: H};
+                const newUrl = new URL(CAPTCHA_HREF + EXAMPLE_SUFFIX);
                 function run() {
                     const tokens = [];
                     for (let i=0; i<testTokenData.length; i++) {
                         tokens[i] = {data: testTokenData[i].data, point: sec1DecodePointFromBytes(testTokenData[i].point), blind: getBigNumFromBytes(testTokenData[i].blind)};
                     }
                     const out = parseRespString(element.string);
                     const xhr = validateAndStoreTokens(newUrl, details.tabId, tokens, out);
-                    before = getMock(TOKEN_COUNT_STR);
+                    before = getMock(bypassTokensCount(1));
                     xhr.onreadystatechange();
-                    after = getMock(TOKEN_COUNT_STR);
+                    after = getMock(bypassTokensCount(1));
                 }
                 setTimeSinceLastResp(0); // reset the variables
-                setMock(TOKEN_COUNT_STR, 0);
-                workflow.__set__("RELOAD_ON_SIGN", false);
-                const newUrl = new URL(EXAMPLE_HREF + EXAMPLE_SUFFIX);
-                expect(consoleMock.error).not.toBeCalled();
-                expect(run).not.toThrow();
-                expect(updateIconMock).toBeCalledTimes(3);
-                expect(updateBrowserTabMock).not.toBeCalled();
-                expect(after == before+testTokenData.length).toBeTruthy();
-                expect(getSpendFlag(newUrl.host)).toBeFalsy();
+                setMock(bypassTokensCount(1), 0);
+                workflow.__with__({"reloadOnSign": () => false})(() => {
+                    const newUrl = new URL(EXAMPLE_HREF + EXAMPLE_SUFFIX);
+                    expect(consoleMock.error).not.toBeCalled();
+                    expect(run).not.toThrow();
+                    expect(updateIconMock).toBeCalledTimes(3);
+                    expect(updateBrowserTabMock).not.toBeCalled();
+                    expect(after == before + testTokenData.length).toBeTruthy();
+                    expect(getSpendFlagMock(newUrl.host)).toBeFalsy();
+                });
             });
         });
 
         describe("test parsing errors", () => {
             test("cannot decode point", () => {
                 function run() {
                     const tokens = [];
-                    for (let i=0; i<testTokens.length; i++) {
-                        tokens[i] = {data: testTokens[i].data, point: sec1DecodePointFromBytes(testTokens[i].point), blind: getBigNumFromBytes(testTokens[i].blind)};
+                    for (let i = 0; i < testTokens.length; i++) {
```

Comment:
My fault again, but it seems like this loop can be refactored since it is used in the same way in the next few functions.